pandas_datareader/data.py

```diff
@@ -581,42 +582,105 @@ def fetch_data(url, name):
     return df
 
 
-_FAMAFRENCH_URL = 'http://mba.tuck.dartmouth.edu/pages/faculty/ken.french/ftp'
+_FAMAFRENCH_URL = 'http://mba.tuck.dartmouth.edu/pages/faculty/ken.french/'
+_FF_PREFIX = 'ftp/'
+_FF_SUFFIX = '_CSV.zip'
 
 
-def get_data_famafrench(name):
-    # path of zip files
-    zip_file_path = '{0}/{1}_TXT.zip'.format(_FAMAFRENCH_URL, name)
+def get_datasets_famafrench():
+    """
+    Get the list of datasets available from the Fama/French data library.
+
+    Returns
+    -------
+    A list of valid inputs for get_data_famafrench.
+    """
+    from bs4 import BeautifulSoup
+
+    with urlopen(_FAMAFRENCH_URL + 'data_library.html') as socket:
+        root = BeautifulSoup(socket.read(), 'html.parser')
+
+    l = filter(lambda x: x.startswith(_FF_PREFIX) and x.endswith(_FF_SUFFIX),
+               [e.attrs['href'] for e in root.findAll('a') if 'href' in e.attrs])
```

Comment:
Check this gist out:
https://gist.github.com/yejianye/5895954

Also, we could pull the Options._parse_url method out into a common function, as it does the importing and error checking for lxml.